### PR TITLE
Switch to HCL 2.0 in the HCL repository

### LINKED ISSUE #473: Switch to hcl v2 from hcl2
See https://github.com/hashicorp/terraform/pull/22982

We also need this migration.

### PR DESCRIPTION
**Pull request by explodingcamera**
Related to https://github.com/hashicorp/terraform/pull/22982 .
This was also mostly just finding and replacing, however I also updated the link in the docs and updated `terraform` to a newer version which also switched to the new hcl repo. Closes #473.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -9,17 +9,17 @@ require (
 	github.com/google/go-cmp v0.3.1
 	github.com/hashicorp/aws-sdk-go-base v0.3.0
 	github.com/hashicorp/go-version v1.2.0
-	github.com/hashicorp/hcl v0.0.0-20180404174102-ef8a98b0bbce // indirect
+	github.com/hashicorp/hcl/v2 v2.0.0
 	github.com/hashicorp/hcl2 v0.0.0-20190821123243-0c888d1241f6
 	github.com/hashicorp/logutils v1.0.0
-	github.com/hashicorp/terraform v0.12.9
+	github.com/hashicorp/terraform v0.12.10-0.20191002221021-39e609d5fd70
```
Comment (wata727) [inline go.mod]: Is there a way to keep this version as 0.12.9?

File: go.mod [hunk 2]
```diff
@@ -9,17 +9,17 @@ require (
 	github.com/google/go-cmp v0.3.1
 	github.com/hashicorp/aws-sdk-go-base v0.3.0
 	github.com/hashicorp/go-version v1.2.0
-	github.com/hashicorp/hcl v0.0.0-20180404174102-ef8a98b0bbce // indirect
+	github.com/hashicorp/hcl/v2 v2.0.0
 	github.com/hashicorp/hcl2 v0.0.0-20190821123243-0c888d1241f6
```
Comment (wata727) [inline go.mod]: Can this dependency be removed?

File: go.mod (hunk 1 above)
Comment (explodingcamera) [reply go.mod]: https://github.com/hashicorp/terraform/commit/39e609d5fd708a3e2c63b347b2890e93b79e06af They switched to the new repo in this commit and some types break without these changes.

**Comment by wata727** [comment]
Thanks @explodingcamera!

**Pull request merged**